### ISSUE TITLE
Enhancing Shared Data Libraries

### ISSUE THREAD

**Issue by ssander5** [bxlab/galaxy-hackathon#30]
**Goal:** 
Design representative testing/tutorial/training data to be published as Shared Data Libraries.  

**Objectives:**
- **Migrate intact Data Libraries from one Galaxy to another** https://github.com/bxlab/galaxy_hackathon/issues/40 https://github.com/bxlab/galaxy_hackathon/issues/24
- Compile and provide small, representative data that is well annotated.
- Aim for data that provides interesting results or highlights typical issues in the analysis
- Aim for a range of workflows
- Base off of published analyses (key publications with reproducible results)
- Publically accessible original data source
- Capture data reduction methods

**Discussion points:**
- Analyses to target? I.e: genomes, transcriptomes/exomes, NGS samples, reference annotation, and any other content required to execute complete workflows.
  - what are the common issues with each analysis to highlight?
- Useful data already on the site?

**Related**
- Submit workflows to ToolShed via the Galaxy workflow interface https://github.com/bxlab/galaxy_hackathon/issues/4  
- Training Data (includes tutorial, example) https://github.com/bxlab/galaxy_hackathon/issues/28
- Coding Hack: Make the necessary modifications to Data Library structure/interface etc.

**Comment by frederikcoppens** [comment]
Where should the "master collection" of these training data sets live? Data libraries on Main? How do we organise this best?

**Comment by ssander5** [comment]
The Workflow Best Practices Team is working on the following according to (https://github.com/bxlab/galaxy_hackathon/issues/29):
- RNA-seq quantification
- RNA-seq variant calling
- ChIP-seq

Would it make sense for us to target the same workflows?  Doing so would link both these initiatives and provide public example sets for learning the best practices....

**Comment by ssander5** [comment]
I started compiling a list of common issues in data for RNA-seq and ChIP-seq (as mentioned above).  These would be things that we would want to consider including in any miniaturized/example data set to include these characteristics.

**RNA-seq data issues:**
- resulting from library prep:
  - adaptor presence 
  - high rRNA persistence 
  - read through/short reads after quality trimming
  - contamination
- resulting from biology of organism:
  - multi-mapping and ambiguity 
  - rare variants

**Chip-seq data issues (not my forte - please chime in!):**
- GC richness bias
- reads in low complexity regions are often discarded
- read density differences between euchromatin and heterochromatin

**Comment by ssander5** [comment]
Another thought on data to maybe search out (I may have leads) - RADtag.  Here are some common issues associated that would be a good idea to also consider.

**RAD-tag data issues:**
- resulting from library construction:
  - weird placement of adapters due to high 5’ end similarity
  - adaptor presence
  - read through/short reads after quality trimming
  - contamination
  - limited sample size
  - single vs double digest
- resulting from biology of organism:
  - SNP calling heterozygotes
  - rare variants
  - mtDNA inclusion
  - multi-mapping and ambiguity

**Comment by frederikcoppens** [comment]
Definitely good idea to link these.

List of issues looks good already. Anyone has datasets? You can upload them through FTP to the Cloud instance or put a link in the comments. We'll add them to the Data Libraries (plan to copy these before we bring the cloud down after the hackathon)

Additional issues:
- wrong species

**Comment by frederikcoppens** [comment]
Suggestion on data reduction procedures? 

Metadata needed per dataset. How to capture?
-   source publication
-   source download
-   datatype
-   species
-   contributor
-   reduction method
-   research area (RNA-seq, ChIP-seq, etc)

**Comment by ssander5** [comment]
By wrong species do you mean that their isn't a reference genome for the species of interest?  Also, I'm working on a list of data that I know of from these types.  I imagine we will have to do some reduction based on our target outcome using command line (i.e.: subset, check for bias, etc.)?

**Comment by ssander5** [comment]
I'm looking through the current list of data that is already up on the galaxy data library to see if there are any good candidates as well

**Comment by cschu** [comment]
Concerning quality trimming: I thought that this is not recommended anymore?http://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-016-0956-2

**Comment by ssander5** [comment]
I had gathered that aggressive trimming without the use of subsequent length filters was the major issue, which is what I was trying to get at with the "short reads after trimming" issue.  Sorry I wasn't clear - also thanks for that source - I hadn't seen it!  

"We found that the majority of the expression changes could be mitigated by imposing a minimum length filter following trimming, suggesting that the differential gene expression was primarily being driven by spurious mapping of short reads."

**Comment by frederikcoppens** [comment]
With wrong species I meant that the data is not the species you expect it to be (due to mix up of samples e.g.). Had several issues like that. 
A more general procedure to tackle this and figure out "contaminations" would be nice too

**Comment by jennaj** [comment]
Shared Libs on Main (http://usegalaxy.org)  == Yes
Shared Libs installed on default Cloudman == Potential 
- some
- all
- in S3 bucket with indexes?
- other options?
- config files
- associated workflows?
- would this be a Galaxy Flavor (training)

**Comment by ssander5** [comment]
Possible dataset for RAD analysis:

**RADseq data from Oaks (chosen because the genome available, has small data sets, unique endpoint analysis for RADseq)**
_Publication:_ Hipp AL et al., "A framework phylogeny of the American oak clade based on sequenced RAD data.",PLoS One, 2014 Apr 4;9(4):e93975

**Libraries:**
- SRR1873227 - 124M
- SRR1873257 - 167M
- SRR1873259 - 172M

**Typical Issues with RADseq Captured:**
- Chloroplast contamination
- Double digest (weird adapter placement - on 3’ side, must remove)
- barcodes still present

**Aims Captured:**
- Small, representative data - Y
- Highlights typical issues -Y
- Published analyses -Y (uses stacks pipeline - test @yvanlebras pipeline?)
- Publicly accessible - Y
- Data reduction methods captured - Y
  - Took only three of the several species for reduced size of analysis
  - No manipulation was required.

**Comment by ssander5** [comment]
I spent time searching for datasets that would be representations that I could simply truncate and still represent common issues. I felt kind of weird about manipulating data for this purpose to make "common problems" occur.  I found a ChIP-seq data set on the Shared Libarary already, so I figured we could leave that at that.  However, I did find a nice set for an RNAseq example library.

Updates:
**ChIP-seq data - already a small test set on Shared Data Library**
I don’t know a lot about ChIP-seq data, but this small test set is probably better than I would find with my limited knowledge.  Let me know if it doesn’t play nice with the new best practices workflow.

**RNA-seq data from Daphnia (Chosen because the genome is available, duplication in genome makes for mapping considerations, three samples of two libraries to allow for cuffdiff/DEseq)** 
_publication:_”StressFlea" (Grant ID 09-EEFG-FP-040, ESF EUROCORES Programme EuroEEFG) 
http://www.esf.org/coordinating-research/eurocores/programmes/euroeefg/projects/stressflea.html

**Libraries:**
- SRR2062534 - 7.3G after truncation
- SRR2062535 - 7.3G after truncation
- SRR2062536 - 7.3G after truncation
- SRR2062779 - 7.3G after truncation
- SRR2062780 - 7.3G after truncation
- SRR2062781 - 7.3G after truncation

I will upload these to the Hack Cloud tomorrow.  I have to download them to my home machine, then the galaxy instance, unless there is a better way to do this from IU’s cluster?

**Typical Issues with RNAseq Captured:**
- Biological:
  - Repetitive genome
  - Rare variants
- Artificial:
  - rRNA enrichment
  - mtDNA enrichment
  - adapter presence

**Aims Captured:**
- Small, representative data - Y 
- Highlights typical issues - Y (see above)
- Published analyses - Y
- Publicly accessible - Y
- Data reduction methods captured - Y 
  - Took first 14,000,000 sequences of each of six libraries to simulate ~1 lane of data which keeping overall size down.  Should still be totally sufficient for an assembly.  I could even down sample more if we wanted a smaller set.  That is not a problem.

**Comment by cschu** [comment]
This is simply awesome.
